packages/bugsnag_flutter/test/channel_client_test.dart

```diff
@@ -124,6 +125,37 @@ void main() {
         expect(channel['createEvent'], hasLength(1));
         expect(channel['createEvent'][0]['deliver'], isTrue);
       });
+
+      test('Metadata is sanitized', () async {
+        final sampleMetadata = {
+          "List": [1, 2, 3],
+          "Enum": BugsnagBreadcrumbType.manual,
+          "Map": Map<String, Object>.from({"1": 2}),
+        };
+
+        client.addMetadata("Hello", sampleMetadata);
+
+        expect(channel['addMetadata'], hasLength(1));
+        final actualMetadata = channel['addMetadata'][0]['metadata'];
+        expect(actualMetadata["List"], equals([1, 2, 3]));
+        expect(actualMetadata["Enum"], equals("BugsnagBreadcrumbType.manual"));
+        expect(actualMetadata["Map"], equals({"1": 2}));
+      });
+
+      test('Breadcrumb metadata is sanitized', () async {
+        final sampleMetadata = {
+          "List": [1, 2, 3],
+          "Enum": BugsnagBreadcrumbType.manual,
+          "Map": Map<String, Object>.from({"1": 2}),
+        };
+        final breadcrumb = BugsnagBreadcrumb("Hello",
+            type: BugsnagBreadcrumbType.manual, metadata: sampleMetadata);
+
+        expect(breadcrumb.metadata?["List"], equals([1, 2, 3]));
+        expect(breadcrumb.metadata?["Enum"],
+            equals("BugsnagBreadcrumbType.manual"));
+        expect(breadcrumb.metadata?["Map"], equals({"1": 2}));
+      });
```

Comment:
It would be good to see these tests run through a full Object -> Json -> Object cycle, which would also catch any future breaks in that contract.